src/pages/components/Overlay.qml

```diff
@@ -486,6 +487,27 @@ Background {
                         pageStack.animatorPush("com.jolla.settings.system.CertificateDetailsPage", {"website": webView.security.subjectDisplayName, "details": webView.security.serverCertDetails})
                     }
                 }
+                onSavePageAsPDF: {
+                    var filename = "unnamed_file"
+                    if (webView.title && webView.title.length !== 0) {
+                        filename = webView.title
+                    } else {
+                        // drop query string from URL and split to sections
+                        var urlSections = webView.url.split("?")[0].split("/")
+                        var leafName = urlSections[urlSections.length - 1]
+                        if (leafName.length !== 0) {
+                            filename = leafName
+                        }
+                    }
+
+                    filename = filename.replace(/[/\\?%*:|"<>]/g, '-').replace(/ /g, '_') + ".pdf"
+                    var targetUrl = DownloadHelper.createUniqueFileUrl(filename, StandardPaths.download)
```

Comment:
filename.replace(/[/\\?%*:|"<>]/g, '-').replace(/ /g, '_') => should be added/moved to the createUniqueFileUrl itself.

If filename has already extension this would concatenate 2nd .pdf e.g. via title "myExample.pdf"